app/controllers/todos_controller.rb

```diff
@@ -0,0 +1,28 @@
+class TodosController < ApplicationController
+  protect_from_forgery with: :null_session
+  def index
+    todos = Todo.all
+    render json: todos
+  end
+
+  def show
+    todo = Todo.find(params[:id])
+    render json: todo
+  end
+
+  def create
```

Comment:
I might add some more testing around this method because a few things:

`create!` is already saving to the database and raising an exception if validations fail. Therefore when you call `.save` you are actually re-saving successful saves or doing nothing in the case of an exception already being raised. In the second case I don't think you are hitting your `else` and instead are just returning a generic unprocessable.

You could switch to something more along these lines that might closer to what you are looking for:

```ruby
todo = Todo.create(params)   # Validations haven't run yet so calling them `valid` might be incorrect. 

if todo.valid?
  render json: todo, status: :ok
else
  render json todo.errors, status: :unprocessable_entity
end
```